gateway/ln-gateway/src/lib.rs

```diff
@@ -813,15 +780,22 @@ impl Gateway {
             .await
             .expect("Gateway configuration should be set");
 
-        let mut federations = Vec::new();
-        for (federation_id, client) in self.clients.read().await.clone() {
-            federations.push(
-                client
-                    .borrow()
-                    .with(|client| self.make_federation_info(client, federation_id))
-                    .await,
-            );
-        }
+        // Minimize the time we hold a lock on the federation manager.
+        let (federations, channels) = {
+            let federation_manager = self.federation_manager.read().await;
+
+            let mut federations = Vec::new();
+            for (federation_id, client) in federation_manager.borrow_clients().clone() {
+                federations.push(
+                    client
+                        .borrow()
+                        .with(|client| self.make_federation_info(client, federation_id))
+                        .await,
+                );
+            }
+
+            (federations, federation_manager.clone_scid_map())
+        };
```

Comment:
Let's make this a function inside `FederationManager`

Reply:
It's not quite that easy because `Gateway::make_federation_info` requires reading from the database. Surely there's a cleaner way to do this though - I dropped a TODO comment there to address this later.